openupgradelib/openupgrade.py

```diff
@@ -863,6 +863,33 @@ def rename_models(cr, model_spec):
                     "new_string": "%s," % new,
                 },
             )
+        # Handle models that reference to this model using reference fields
+        cr.execute("""
+             SELECT model, name
+             FROM ir_model_fields
+             WHERE ttype='reference'
+             """)
+        rows = cr.fetchall()
+        for row in rows:
+            table = row[0].replace('.', '_')
```

Comment:
This is still not correct, as not all models has that table name. Maybe you can trace from the field to the table that hosts it.

Reply:
IMO, I have three reasons to proceed this way, and I think you should accept:

 - I am doing the same as https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L783. If that line is acceptable in the same method, mine should also be acceptable.
 - Moreover, I am calling a "table_exists" to check if that table exists.
 - Moreover, nobody uses reference fields in custom modules. Only few models in odoo community have this kind of field and all of them their table is the same as their model name changing the `.` by `_`.

An, no, there is no way to trace the table from the field and model name. Only environment could help.

Reply:
> I am doing the same as https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L783. If that line is acceptable in the same method, mine should also be acceptable.

Such replacement is not for table names, but for certain identifiers, so please don't mix things.

I'm only telling you that if there's a way to get the real table name, we should use it.

Reply:
@StefanRijnhart do you see this reasonable? Maybe we should have a table mapping for known standard changes (like act_window and company).

Reply:
But here I am not doing any table renames, I don't understand what are you saying :S

Reply:
Ah yes, forget about that. I thought we were talking about the table of the model that was being renamed. In this case the check on table_exists is fine by me.

Reply:
But the `ir.actions.*` models don't have any reference field. And I doubt any custom modules added one. And if a custom module added one, they are records with external id, thus they will be updated during load.

Thus, I don't think such mapping is needed when the model in question is not affected.

Reply:
I tend to agree in this case. Maybe at some point in the future we can create a convenience method in the lib to provide a model's table based on the model name plus a lookup table for that rare list of exceptions?

Reply:
OK, let's go with this.